internal/observability/middleware_test.go

```diff
@@ -589,6 +594,89 @@ func TestMiddlewareLoggingWithApplicationErrorConfiguration(t *testing.T) {
 	}
 }
 
+func TestMiddlewareLoggingDifferentCallerProcedureWithSameEdge(t *testing.T) {
+	defer stubTime()()
+	req := &transport.Request{
+		Caller:    "caller",
+		Service:   "service",
+		Encoding:  "raw",
+		Procedure: "procedure",
+	}
+
+	baseFields := func() []zapcore.Field {
+		return []zapcore.Field{
+			zap.String("source", req.Caller),
+			zap.String("dest", req.Service),
+			zap.String("transport", unknownIfEmpty(req.Transport)),
+			zap.String("procedure", req.Procedure),
+			zap.String("encoding", string(req.Encoding)),
+			zap.String("routingKey", req.RoutingKey),
+			zap.String("routingDelegate", req.RoutingDelegate),
+		}
+	}
+
+	infoLevel := zapcore.InfoLevel
+	warnLevel := zapcore.WarnLevel
+
+	core, logs := observer.New(zapcore.DebugLevel)
+	mw := NewMiddleware(Config{
+		Logger:           zap.New(core),
+		Scope:            metrics.New().Scope(),
+		ContextExtractor: NewNopContextExtractor(),
+		Levels: LevelsConfig{
+			Default: DirectionalLevelsConfig{
+				Success:          &infoLevel,
+				ApplicationError: &warnLevel,
+				// Leave failure level as the default.
+			},
+		},
+	})
+
+	getLog := func(t *testing.T) observer.LoggedEntry {
+		entries := logs.TakeAll()
+		e := entries[len(entries)-1]
+		e.Entry.Time = time.Time{}
+		return e
+	}
+
+	wantFields := []zapcore.Field{
+		zap.Duration("latency", 0),
+		zap.Bool("successful", true),
+		zap.Skip(), // ContextExtractor
+	}
+
+	for i := 0; i < 2; i++ {
+		req.CallerProcedure = fmt.Sprintf("caller-procedure-%d", i)
+
+		t.Run(fmt.Sprintf("unary_inbound_%d", i), func(t *testing.T) {
+			err := mw.Handle(
+				context.Background(),
+				req,
+				&transporttest.FakeResponseWriter{},
+				fakeHandler{},
+			)
+			assert.NoError(t, err, "Unexpected error from middleware.")
+			logContext := append(
+				baseFields(),
+				zap.String("direction", string(_directionInbound)),
+				zap.String("sourceProcedure", req.CallerProcedure),
+				zap.String("rpcType", "Unary"),
+			)
```

Comment:
Should the fields `baseFields()`, `direction` and  `rpcType` be in `wantFields`?